# HG changeset patch
# User jwe
# Date 749250840 0
#      Tue Sep 28 21:14:00 1993 +0000
# Node ID f8e230ec7745f18f1ddd75fbec43eef3287e3613
# Parent  0869dc33276c1ccd9c4608cad81776dc56740556
[project @ 1993-09-28 21:14:00 by jwe]
(QPSOL::minimize (double&, int&, Vector&)): Correct dimensions of work
arrays, based on QPSOL source.
Pass ncon, which is max (1, nclin), not n, as the row dimension of
clin.

diff --git a/liboctave/QPSOL.cc b/liboctave/QPSOL.cc
--- a/liboctave/QPSOL.cc
+++ b/liboctave/QPSOL.cc
@@ -137,31 +137,30 @@ QPSOL::minimize (double& objf, int& info
   int *istate = new int [nctotl];
 
   double *px = x.fortran_vec ();
 
   int iter = 0;
   lambda.resize (nctotl);
   double *pclambda = lambda.fortran_vec ();
 
-  int tmp = n < nclin ? n : nclin;
-  int leniw = n + 2 + tmp;
+  int leniw = 2 * n;
 
   int lenw;
   int ncon = nclin > 1 ? nclin : 1;
   if (lp == 0 || nclin >= n)
     lenw = 2*n*(n + 2) + nclin + 2*ncon;
   else
     lenw = 2*ncon*(1 + ncon) + 4*n + nclin;
 
   int *iw = new int [leniw];
   double *w = new double [lenw];
 
-  F77_FCN (qpsol) (&itmax, &msglvl, &n, &nclin, &nctotl, &n, &n, &n,
-		   &bigbnd, pa, pbl, pbu, pc, featol, ph, qphess,
+  F77_FCN (qpsol) (&itmax, &msglvl, &n, &nclin, &nctotl, &ncon, &n,
+		   &n, &bigbnd, pa, pbl, pbu, pc, featol, ph, qphess,
 		   &cold, &lp, &orthog, istate, px, &inform, &iter,
 		   &objf, pclambda, iw, &leniw, w, &lenw);
 
   delete [] featol;
   delete [] istate;
   delete [] iw;
   delete [] w;
 
